sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorEventSource.cs

```diff
@@ -0,0 +1,395 @@
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that the partition processing task is stopping.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is being stopped.</param>
+        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
+        ///
+        [Event(17, Level = EventLevel.Verbose, Message = "Stopping partition processing task for partition id '{0}' with reason '{1}'")]
+        public void StopPartitionProcessing(string partitionId, ProcessingStoppedReason reason)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(17, partitionId, reason);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while stopping a partition processing task.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition.</param>
+        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        [Event(18, Level = EventLevel.Error, Message = "An exception occurred while stopping a partition processing task for partition id '{0}' with reason '{1}'. (Error Message: '{2}')")]
+        public void StopPartitionProcessingError(string partitionId, ProcessingStoppedReason reason, string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(18, partitionId, reason, errorMessage ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that the partition processing task has been stopped.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is being stopped.</param>
+        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
+        ///
+        [Event(19, Level = EventLevel.Verbose, Message = "Stopped partition processing task for partition id '{0}' with reason '{1}'")]
+        public void StoppedPartitionProcessing(string partitionId, ProcessingStoppedReason reason)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(19, partitionId, reason);
+            }
+        }
+
+        /// <summary>
+        ///   Signals the <see cref="EventProcessorClient" /> to stop processing events.
+        /// </summary>
+        ///
+        /// <param name="identifier">A unique name used to identify the event processor.</param>
+        ///
+        [Event(20, Level = EventLevel.Verbose, Message = "Stop processing events. (Identifier '{0}')")]
+        public void EventProcessorStop(string identifier)
```

Comment:
Was this intended as a `Start` event pair with `EventProcessorStopCompleted`? If so I believe they should have adjacent event Ids and have a name like `EventProcessorStopStart`.